pkg/kubelet/winstats/perfcounters.go

```diff
@@ -0,0 +1,120 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package winstats
+
+import (
+	"errors"
+	"fmt"
+	"github.com/lxn/win"
+	"time"
+	"unsafe"
+)
+
+const (
+	CPUQuery                  = "\\Processor(_Total)\\% Processor Time"
+	MemoryPrivWorkingSetQuery = "\\Process(_Total)\\Working Set - Private"
+	MemoryCommittedBytesQuery = "\\Memory\\Committed Bytes"
+)
+
+// Metric collected from a counter
+type Metric struct {
+	Name  string
+	Value float64
+}
+
+// NewMetric construct a Metric struct
+func NewMetric(name string, value float64) Metric {
+	return Metric{
+		Name:  name,
+		Value: value,
+	}
+}
+
+func (metric Metric) String() string {
+	return fmt.Sprintf(
+		"Name: %s | Value: %s",
+		metric.Name,
+		metric.Value,
+	)
+}
+
+func readPerformanceCounter(counter string, sleepInterval int) (chan Metric, error) {
+
+	var queryHandle win.PDH_HQUERY
+	var counterHandle win.PDH_HCOUNTER
+
+	ret := win.PdhOpenQuery(0, 0, &queryHandle)
```

Comment:
Not currently. The perf counters continue to collect data for as long the kubelet process is running via a [go routine](https://github.com/bobbypage/kubernetes/blob/efc38607b27aee8c505c6f2b88798878c2c11eaf/pkg/kubelet/winstats/perfcounters.go#L85-L87). Is it necessary to close it? If so, where should that close happen since the handle is used for as long as the kubelet is running?